### PR TITLE
Adding button stats tables

### PR DESCRIPTION
**Pull request by blackshadowshade**
This is a working pull request made for convenience so that everyone can see where we're up to.

Addresses #614.

Requires database updates 00614_stats_tables_01.sql and 00614_stats_tables_02.sql

### CODE REVIEW

**Comment by blackshadowshade** [comment]
Now that a database update script also exists for AMP stacks, I'll need to update that too.

**Comment by blackshadowshade** [comment]
Right, I've updated the script to add stats initialisation now.

**Comment by blackshadowshade** [comment]
http://jenkins.buttonweavers.com:8080/job/buttonmen-blackshadowshade/620/ (assuming it passes)

File: deploy/database/drop_databases.sql [hunk 1]
```diff
@@ -1,2 +0,0 @@
-DROP DATABASE IF EXISTS buttonmen;
-DROP DATABASE IF EXISTS buttonmen_test;
```
Comment (cgolubi1) [inline deploy/database/drop_databases.sql]: Why does this pull remove the drop_databases.sql script?  Is that intentional?  Do you need to merge from upstream?

**Comment by cgolubi1** [comment]
Other than that, this looks straightforward, and the only test i need to do is that the regular and update SQL files yield the same result as each other.

File: deploy/database/drop_databases.sql (hunk 1 above)
Comment (blackshadowshade) [inline deploy/database/drop_databases.sql]: If you look at the function recreate_database() in database_initialise.php, you'll see that the drop actually happens there. Thus, we don't need drop_databases.sql any more for the AMP stacks.

You also don't seem to use it in your initialise_schema script, so it's orphaned, as far as I can tell.

Comment (cgolubi1) [inline deploy/database/drop_databases.sql]: It's not orphaned --- i never used it, and you introduced it recently for use by AMP, which is why i was surprised that you removed it.  But, anyway, i have no objection.

**Comment by cgolubi1** [comment]
Looks fine.

**Pull request merged**